### PR TITLE
Creates a discoverable rqt plugin within the package

### PR DESCRIPTION
**Pull request by Voldivh**
This PR creates the initial package structure with a discoverable plugin.

Follow the next commands (inside the workspace where you cloned the repository) in order to test the plugin:
```bash
colcon build --packages-select rqt_gauges_2
source install/setup.bash
ros2 run rqt_gauges_2 rqt_gauges_2
```

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

### CODE REVIEW

**Comment by francocipollone** [comment]
Nice! One comment, I noticed you are using `catkin` and `rospy` instead of `rclpy` so it wouldn't be targeting ros2. How do you plan to do the migration? Just to get the idea.

**Comment by Voldivh** [comment]
> Nice! One comment, I noticed you are using `catkin` and `rospy` instead of `rclpy` so it wouldn't be targeting ros2. How do you plan to do the migration? Just to get the idea.

Forgot to delete the CMakeLists.txt file, it is not needed catkin nor rospy.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    maintainer_email='eloy.briceno@ekumenlabs.com',
+    description=(
+        'rqt_gauges_2 is a Python GUI plugin providing a visualization tool for several sensors.'
+    ),
+    entry_points={
+        'console_scripts': [
+            'rqt_gauges_2 = ' + package_name + '.main:main',
+        ],
+    },
+)
```
Comment (francocipollone) [inline setup.py]: nit: missing empty line at the end of the file

File: src/rqt_gauges_2/gauges_2.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            self._widget.setWindowTitle(
+                self._widget.windowTitle() + (' (%d)' % context.serial_number()))
+        self._context.add_widget(self._widget)
+
+    def shutdown_console_widget(self):
+        if self._console_widget is not None and hasattr(self._console_widget, 'shutdown'):
+            self._console_widget.shutdown()
+
+    def shutdown_plugin(self):
+        self.shutdown_console_widget()
```
Comment (francocipollone) [inline src/rqt_gauges_2/gauges_2.py]: nit: missing empty line at the end of the file

File: src/rqt_gauges_2/main.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+from rqt_gui.main import Main
+
+
+def main():
+    main = Main()
+    sys.exit(main.main(sys.argv, standalone='rqt_gauges_2.gauges_2.Gauges2'))
+
+
+if __name__ == '__main__':
+    main()
```
Comment (francocipollone) [inline src/rqt_gauges_2/main.py]: nit: missing empty line at the end of the file

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[develop]
+script_dir=$base/lib/rqt_gauges_2
+[install]
+install_scripts=$base/lib/rqt_gauges_2
```
Comment (francocipollone) [inline setup.cfg]: nit: missing empty line at the end of the file

File: resource/gauges_2_widget.ui [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    </rect>
+   </property>
+   <property name="text">
+    <string>Original Name</string>
+   </property>
+  </widget>
+ </widget>
+ <resources/>
+ <connections/>
+</ui>
```
Comment (francocipollone) [inline resource/gauges_2_widget.ui]: nit: missing empty line at the end of the file

File: plugin.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    <description>
+      A Visualization tool to keep track of the values of sensors
+    </description>
+    <qtgui>
+      <label>Visualization</label>
+      <icon type="theme">applications-other</icon>
+      <statustip>Visualization tool.</statustip>
+    </qtgui>
+  </class>
+</library>
```
Comment (francocipollone) [inline plugin.xml]: nit: missing empty line at the end of the file

File: package.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  <name>rqt_gauges_2</name>
+  <version>0.0.0</version>
+  <description>Visualization plugin for several sensors.</description>
+
+  <maintainer email="eloy.briceno@ekumenlabs.com">Eloy Briceno</maintainer>
+
+  <!-- One license tag required, multiple allowed, one license per tag -->
+  <!-- Commonly used license strings: -->
+  <!--   BSD, MIT, Boost Software License, GPLv2, GPLv3, LGPLv2.1, LGPLv3 -->
+  <license>TODO</license>
```
Comment (francocipollone) [inline package.xml]:
```suggestion
  <license file="LICENSE">BSD Clause 3</license>
```

**Review by francocipollone — CHANGES_REQUESTED** [review]
LGTM! Some minor comments!

File: package.xml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>rqt_gauges_2</name>
+  <version>0.0.0</version>
```
Comment (agalbachicar) [inline package.xml]: Nit: probably you want 0.0.1

File: setup.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
+from setuptools import setup
+
+package_name = 'rqt_gauges_2'
+
+setup(
+    name=package_name,
+    version='0.0.0',
```
Comment (agalbachicar) [inline setup.py]: Nit: probably you want 0.0.1

File: package.xml (hunk 7 above)
Comment (agalbachicar) [reply package.xml]: And I think you need to add the license file too.

**Review by agalbachicar — COMMENTED** [review]
I think you also need to configure the python linters.

File: setup.py (hunk 1 above)
Comment (Voldivh) [reply setup.py]: Done.

File: src/rqt_gauges_2/gauges_2.py (hunk 2 above)
Comment (Voldivh) [reply src/rqt_gauges_2/gauges_2.py]: Done.

File: src/rqt_gauges_2/main.py (hunk 3 above)
Comment (Voldivh) [reply src/rqt_gauges_2/main.py]: Done.

File: setup.cfg (hunk 4 above)
Comment (Voldivh) [reply setup.cfg]: Done.

File: resource/gauges_2_widget.ui (hunk 5 above)
Comment (Voldivh) [reply resource/gauges_2_widget.ui]: Done.

File: plugin.xml (hunk 6 above)
Comment (Voldivh) [reply plugin.xml]: Done.

File: package.xml (hunk 7 above)
Comment (Voldivh) [reply package.xml]: Done.

File: package.xml (hunk 8 above)
Comment (Voldivh) [reply package.xml]: Done.

File: setup.py (hunk 9 above)
Comment (Voldivh) [reply setup.py]: Done.

File: plugin.xml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<library path="src">
+  <class name="Gauges2" type="rqt_gauges_2.gauges_2.Gauges2" base_class_type="rqt_gui_py::Plugin">
+    <description>
+      A Visualization tool to keep track of the values of sensors
+    </description>
+    <qtgui>
+      <label>Visualization</label>
+      <icon type="theme">applications-other</icon>
+      <statustip>Visualization tool.</statustip>
+    </qtgui>
```
Comment (francocipollone) [inline plugin.xml]: You can add a group so the plugin is located under the folder.


Also use Gauges for the plugin name.

```suggestion
    <qtgui>
      <group>
        <label>Visualization</label>
        <icon type="theme">folder</icon>
        <statustip>Plugins related to visualization.</statustip>
      </group>
      <label>Gauges</label>
      <icon type="theme">applications-other</icon>
      <statustip>Gauges visualization tool.</statustip>
    </qtgui>
```

Comment (francocipollone) [reply plugin.xml]: Based on rqt_image_view: https://github.com/ros-visualization/rqt_image_view/blob/master/plugin.xml

Check it using 
```
rqt
```

And then going to the `plugins` "tab"(drop down list).

Comment (Voldivh) [reply plugin.xml]: Done.

**Review by francocipollone — APPROVED** [review]
LGTM

**Pull request merged**